mdal/mdal_data_model.hpp

```diff
@@ -38,6 +38,137 @@ namespace MDAL
 
   typedef std::vector< std::pair< std::string, std::string > > Metadata;
 
+  class DateTime;
+
+  class Duration
+  {
+    public:
+      enum Unit
+      {
+        milliseconds = 0,
+        seconds,
+        minutes,
+        hours,
+        days,
+        weeks
+      };
+
+      Duration();
+
+      ///TODO : implementation of operator, copy constructor, copy assignment, move operation
+
+      Duration( double duration, Unit unit );
+
+      double value( Unit unit ) const;
+
+    private:
+      Duration( int64_t ms );
+      int64_t mDuration; //in ms
+
+      friend class DateTime;
+  };
+
+  class DateTime
+  {
+    public:
+
+      //! Defaul constructor
+      DateTime(): mValid( false )
+      {}
+      //! Copy constructor
+      DateTime( const DateTime &other ): mJulianTime( other.mJulianTime )
+      {}
+
+      //static method returning DateTime
+
+      //! Create an instance from gregorian/julian calendar (or standart)
+      static DateTime fromStandartValue( int year, int month, int day, int hours = 0, int minutes = 0, double seconds = 0 );
+
+      //! Create an instance from julian day
+      static DateTime fromJulianDay( double julianDay )
+      {
+        return DateTime( int64_t( julianDay * 24 * 3600 * 1000 ) );
+      }
+
+      //! Create an default instance
+      static DateTime fromDefault()
+      {
+        return DateTime();
+      }
+
+      //! Returns a string with the date/time expressed in Greogrian/Julian calendar with ISO8601 format (local time zone)
```

Comment:
I think we do not need to use local time zone for output. MDAL API should return universal gregorian UTC time string

Reply:
I am not sure... My point of view :
Returning explicitly mean that the time is surely UTC. But, often, that's not the case and the time zone is unknown. 
When the time zone is not specified, that's mean the time is local (https://stackoverflow.com/questions/33114386/datetime-iso-8601-without-timezone-component).
In this implementation, time zone is not supported, so MDAL "doesn't know" the zone zone and return the time in local time.
If we decide that MDAL supports time zone (this is another question), if time zone is known, MDAL should return the time with time zone or UTC time, in conformity with ISO8601. If the tim zone is unknown, MDAL should return in local time.
Open for discussion...

Reply:
i would keep it as simple as possible